src/sentry/lang/javascript/processor.py

```diff
@@ -997,31 +990,13 @@ def process_frame(self, processable_frame, processing_task):
                             {"type": EventError.JS_MISSING_SOURCE, "url": http.expose_url(abs_path)}
                         )
 
-                # the tokens are zero indexed, so offset correctly
-                new_frame["lineno"] = token.src_line + 1
-                new_frame["colno"] = token.src_col + 1
-
-                # Try to use the function name we got from symbolic
-                original_function_name = token.function_name
-
-                # In the ideal case we can use the function name from the
-                # frame and the location to resolve the original name
-                # through the heuristics in our sourcemap library.
-                if original_function_name is None:
-                    last_token = None
-
-                    # Find the previous token for function name handling as a
-                    # fallback.
-                    if (
-                        processable_frame.previous_frame
-                        and processable_frame.previous_frame.processor is self
-                    ):
-                        last_token = processable_frame.previous_frame.data.get("token")
-                        if last_token:
-                            original_function_name = last_token.name
+                # The tokens are 1-indexed.
+                new_frame["lineno"] = token.line
+                new_frame["colno"] = token.col
 
-                if original_function_name is not None:
-                    new_frame["function"] = original_function_name
+                # If we have a usable function name from symbolic or we have no initial function name at all
+                if token.function_name not in useless_fn_names or not new_frame.get("function"):
+                    new_frame["function"] = token.function_name
 
                 filename = token.src
                 # special case webpack support
```

Comment:
maybe moving this out to a separate function makes the intent clearer here, maybe even with some unit tests.